pkg/config/engine/docker/docker.go

```diff
@@ -128,6 +129,28 @@ func (c Config) Save(path string) (int64, error) {
 		return 0, fmt.Errorf("unable to convert to JSON: %v", err)
 	}
 
-	n, err := engine.Config(path).Write(output)
+	n, err := config.Raw(path).Write(output)
 	return int64(n), err
 }
+
+// GetRuntime returns the TOML object of the runtime passed as input
+func (c *Config) GetRuntime(name string) (interface{}, error) {
+	if c == nil {
+		return nil, fmt.Errorf("config is nil")
+	}
+
+	config := *c
+
+	var runtimes map[string]interface{}
+	if _, exists := config["runtimes"]; exists {
+		runtimes = config["runtimes"].(map[string]interface{})
+	} else {
+		return nil, fmt.Errorf("config is nil")
+	}
+
+	if r, ok := runtimes[name]; ok {
```

Comment:
Looking at the code above, I think that `runtimes` could be nil.

Reply:
@tariq1890 one suggestion I have for improving the readability here is to invert the previous if statement so that we return early if no runtimes exist in the config. I had similar concerns as Evan when reading this the first time.

Reply:
I've refactored this function. It should look a lot cleaner now. I've made it more concise eliminating the redundant return statements. Please check now